### ISSUE TITLE
Question: How to develop and include a custom plugin

### ISSUE THREAD

**Issue by Noranius** [xBimTeam/XbimWindowsUI#170]
We would like to add a custom plugin to show visualization depending on related tasks and processes. However, I could not find any documentation about that and re engineering that from the code only would be quite time consuming. Is it possible to develop and include a self-written plugin into the xBIM Xplorer?

**Comment by CBenghi** [comment]
Hello, 

the plugin solution is half baked and we don't have much documentation on that.
What type of APIs would you need to have access to?

Switching entities on/off is not super trivial with the current layer-styler since entities are grouped, but you can easily write you own one that does not group entities to make show/hide easier.

**Comment by Noranius** [comment]
Exactly, I would need access to hide and show entities. Later on, it would be interesting to have access to generate entities. I could have done all this stuff by implementing my own layer-styler, but I thought that this stuff would be additionally to xBIM, hence, implementing an addon sounds better to me. However, if the plugin system is not available I can directly extend the xBIM Xplorer as well.
